airbyte-cdk/python/connector_builder/connector_builder_handler.py

```diff
@@ -2,19 +2,47 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
 
+import dataclasses
 from datetime import datetime
+from typing import Any, Mapping
 
-from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage, Type
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.models import Type
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.declarative.declarative_source import DeclarativeSource
 from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
 from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from connector_builder.message_grouper import MessageGrouper
 
 
 def list_streams() -> AirbyteMessage:
     raise NotImplementedError
 
 
-def stream_read() -> AirbyteMessage:
-    raise NotImplementedError
+DEFAULT_MAXIMUM_NUMBER_OF_PAGES_PER_SLICE = 5
+DEFAULT_MAXIMUM_NUMBER_OF_SLICES = 5
+DEFAULT_MAX_RECORDS = 100
+
+
+def read_stream(source: DeclarativeSource, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog) -> AirbyteMessage:
+    try:
+        if "__test_read_config" not in config:
```

Comment:
nit: Feels like this should be optional since we're not requiring any of the keys to be present, and are handling the case where it isn't set.

Reply:
agreed. fixed